src/miniscript/context.rs

```diff
@@ -400,6 +447,105 @@ impl ScriptContext for Segwitv0 {
         // The witness stack cost is the first element of the tuple
         ms.ext.max_sat_size.map(|x| x.0)
     }
+
+    fn pk_len<Pk: MiniscriptKey>(_pk: &Pk) -> usize {
+        34
+    }
+}
+
+/// Tap ScriptContext
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum Tap {}
+
+impl ScriptContext for Tap {
+    fn check_terminal_non_malleable<Pk: MiniscriptKey, Ctx: ScriptContext>(
+        _frag: &Terminal<Pk, Ctx>,
+    ) -> Result<(), ScriptContextError> {
+        // No fragment is malleable in tapscript context.
+        // Certain fragments like Multi are invalid, but are not malleable
+        Ok(())
+    }
+
+    fn check_witness<Pk: MiniscriptKey, Ctx: ScriptContext>(
+        witness: &[Vec<u8>],
+    ) -> Result<(), ScriptContextError> {
+        if witness.len() > MAX_STACK_SIZE {
+            return Err(ScriptContextError::MaxWitnessItemssExceeded);
+        }
+        Ok(())
+    }
+
+    fn check_global_consensus_validity<Pk: MiniscriptKey, Ctx: ScriptContext>(
+        ms: &Miniscript<Pk, Ctx>,
+    ) -> Result<(), ScriptContextError> {
+        // No script size checks for global consensus rules
+        // Should we really check for block limits here.
+        // When the transaction sizes get close to block limits,
+        // some guarantees are not easy to satisfy because of knapsack
+        // constraints
```

Comment:
I think it'd be reasonable to enforce here 50Kb limit, which is 1/2 the max transaction size according to mempool standardness rules (100kb) so saves healthy margin for other interesting stuff. "50kb should be enough for everybody". Miniscript doesn't care to target non standard txs, right?

Reply:
ah "global consensus" v.s. local.